designs/add-recap-search-option-about-representation/readme.md

```diff
@@ -0,0 +1,113 @@
+**Proposed Change: Add a RECAP search option about representation.**
+
+*This proposal stems from freelawproject/courtlistener issue #[806](https://github.com/freelawproject/courtlistener/issues/806).*
+
+**CONTENTS**
+- Problem statement
+- Proposed solution by component
+  - User Interface
+  - Django code for parsing and storing docket data
+  - Solr
+  - Already-stored docket data
+  - Documentation
+- Test cases
+
+# Problem statement
+
+PACER and RECAP search users do not currently have an option to search for dockets where at least one petitioner had representation. Such dockets generally have higher quality complaints, motions, briefs, citations, and prose. Because that option is not available, searchers have to use other means that cost time and money to differentiate such cases from those where every petitioner is pro se.
+
+# Proposed solution 
+
+Enable a RECAP search option for retrieving dockets where at least one petitioner had representation.
+
+Enabling the option requires changes in (1) the user interface, (2) the server-side Django code that parses and stores docket data, (3) the Solr search configuration, (4) the already-stored docket data, and (5) documentation.
+
+The RECAP browser extension will not need to be changed.
+
+## Proposed solution: User Interface
+
+On the CourtListener.com [RECAP search page](https://www.courtlistener.com/recap/), **add** a select list to the search options.  The list's label will be: "At least one petitioner is represented".
+
+**Provide** three select list options.  The first empty/blank value will be the default.
+- ""
+- "Yes"
+- "No (Every party is pro se)"
+
+Also, **add** a pop-up help tip, perhaps through a question-mark icon, with this text:
+"Parties treated as petitioners include plaintiffs, appellants, applicants, complainants, claimants, and relators. If a party's name does not match the corresponding attorney's name, then the party is assumed to have representation, though this may not always be true."
+
+## Proposed solution:  Django code for parsing and storing docket data
+
+**Modify** the existing Django code that is for parsing and storing docket data.  Start with the code that contains a representation of the data model.
+
+Within that code's representation of a docket, **add** a boolean attribute that will be used to identify whether the docket has a petitioner that is represented or not.  For that attribute, use a name like `petitioner represented`.  (Format the name with capitals, underscores, etc. as appropriate.)
+
+The Django code will, in effect, change the data model and storage just like the following SQL would change the `search_docket` table.  (This SQL is only to illustrate what the Django script will do.  This SQL will not be part of the proposed solution.)
+
+``` sql
+ALTER TABLE search_docket ADD COLUMN petitioner_represented BOOLEAN DEFAULT 'f';
```

Comment:
When you're ready to do a DB migrations, the documentation for that is here: 

https://docs.djangoproject.com/en/2.0/topics/migrations/